### DISCUSSION TITLE
Azure account name doesn't update if I switch subscriptions

### DISCUSSION THREAD

**Discussion by justin-newman** [JanDeDobbeleer/oh-my-posh#726] [General]
Is there something I'm missing to get this to change based one which one is active?
![image](https://user-images.githubusercontent.com/23383538/119023288-e2ea4380-b95e-11eb-9d88-4896cf35668c.png)

**Comment by justin-newman** [comment]
oh I think I figured it out. it's just using the az module instead of az cli

**Reply by justin-newman** [reply]
while you are looking into the other question about using the account name, it would also be nice to be able to choose which tool to use.

**Reply by JanDeDobbeleer** [reply]
Good idea!

**Reply by justin-newman** [reply]
except I took a look at the code for the az segment and it looks like it is supposed to be using the az cli instead of the az powershell module. Not sure what is up with that though.

**Reply by justin-newman** [reply]
It looks like the AZ module will set the environment variables when you change subscriptions, but it doesn't do that with the cli. I see two separate functions on that page which should both work respectively if you were to add the variable to choose between the cli or the powershell module. 

If I set $env:Az_enabled = $false, it correctly picks up the environment set with the az account show command.

Sorry if I'm annoying you with comments. I am just trying to get you as much info as I can to help make it easier to update.

![image](https://user-images.githubusercontent.com/23383538/119164142-56e92200-ba19-11eb-8480-6c67e88523ae.png)

**Reply by JanDeDobbeleer** [reply]
Btw, I have a release coming up that disables the module by default. That should fix this.

**Reply by JanDeDobbeleer** [reply]
The latest release sets that variable to false by default.

**Reply by justin-newman** [reply]
nice, yep that seems to have fixed the default read
